planner/core/logical_plan_builder.go

```diff
@@ -2585,7 +2585,9 @@ func (b *PlanBuilder) buildUpdate(update *ast.UpdateStmt) (Plan, error) {
 
 	updt := Update{OrderedList: orderedList}.Init(b.ctx)
 	updt.SetSchema(p.Schema())
-	updt.SelectPlan, err = DoOptimize(b.optFlag, p)
+	// We cannot apply projection elimination when building the subplan, because
+	// columns in orderedList cannot be resolved.
+	updt.SelectPlan, err = DoOptimize(b.optFlag&^flagEliminateProjection, p)
```

Comment:
I'm not sure whether there will be some bugs if we disable the projection elimination rule. As you mentioned in the PR description that "The root cause of the error is that during projection elimination, we do not resolve columns in `OrderedList` field of `Update`", can we fix this issue in the projection elimination rule for the `Update` operator?

Reply:
Then we need to pass an additional parameter of type `Assignment` to `DoOptimize`, it is pretty ugly and ad-hoc IMHO.